scipy/fft/_pocketfft/setup.py

```diff
@@ -0,0 +1,74 @@
+
+# TODO: As of python 3.6, CCompiler has a `has_flag` method to use instead.
+def has_flag(compiler, flagname):
+    # Return True if the flag is supported by the compiler
+    import tempfile
+    from distutils.errors import CompileError
+    with tempfile.NamedTemporaryFile('w', suffix='.cpp') as f:
+        f.write('int main (int argc, char **argv) { return 0; }')
+        try:
+            compiler.compile([f.name], extra_postargs=[flagname])
+        except CompileError:
+            return False
+    return True
+
+
+def get_std_flag(compiler):
+    # Test the compiler for the highest available c++ standard flag
+    if compiler.compiler_type == 'msvc':
+        flags = ['/std:c++17', '/std:c++14', None]
+    else:
+        flags = ['--std=c++17', '--std=c++14', '--std=c++11']
```

Comment:
The Intel compiler flag is platform dependent IIRC, uses MSVC-like `/xxx` on Windows and regular (`-`) flags on other platforms. See note under the table in https://software.intel.com/en-us/articles/c17-features-supported-by-intel-c-compiler and `numpy/distutils/intelccompiler/py`.

Reply:
`intelw` added. I can't test it but the [docs](https://software.intel.com/en-us/cpp-compiler-developer-guide-and-reference-std-qstd) say the form is `/Qstd=c++##` on windows.